web-common/src/features/dashboards/hooks/useCellInspector.ts

```diff
@@ -0,0 +1,101 @@
+import { cellInspectorStore } from "../stores/cellInspectorStore";
+
+export function useCellInspector() {
+  let hoverTimeout: number | null = null;
+  let lastHoveredValue: string | null = null;
+  let isOpen = false;
+
+  const inspectCell = (value: string, event: MouseEvent, persist = false) => {
+    if (!value) return;
+
+    const stringValue = String(value);
+
+    if (persist) {
+      isOpen = true;
+      cellInspectorStore.open(stringValue, {
+        x: event.clientX,
+        y: event.clientY,
+      });
+    } else if (!isOpen) {
+      cellInspectorStore.open(stringValue, {
+        x: event.clientX,
+        y: event.clientY,
+      });
+    }
+  };
+
+  const handleMouseEnter = (value: string, e: MouseEvent) => {
+    // Clear any existing timeout
+    if (hoverTimeout) {
+      clearTimeout(hoverTimeout);
+      hoverTimeout = null;
+    }
+
+    // Only track the hovered value but don't open the inspector
+    hoverTimeout = window.setTimeout(() => {
+      if (value !== lastHoveredValue) {
+        lastHoveredValue = value;
+        // Store the value but don't open the inspector
```

Comment:
This comment is a little weird since the code below only executes when the inspector is actually open. Worth a revisit.